### PR TITLE
RELATED: FET-903 simplify and clean up auth-related code

### PR DESCRIPTION
**Pull request by no23reason**
Go through the auth-related code in tiger and bear and simplify it/clean it up. The transformations were meant to maintain the original behaviour as much as possible. The transformations include:
* using async if it makes the code simpler
* avoiding separate error handlers that can be merged
* removing boilerplate-y code especially in api-client-ber user section

The transformations were done one-by-commit in bear and then applied in bulk to tiger so go commit by commit for some more info on a given transformation.

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: libs/api-client-bear/src/user.ts [hunk 1]
```diff
@@ -188,25 +156,23 @@ export class UserModule {
...
-            return {
-                login: bootstrapResource.accountSetting.login,
-                loginMD5: bootstrapResource.current.loginMD5,
-                firstName: bootstrapResource.accountSetting.firstName,
-                lastName: bootstrapResource.accountSetting.lastName,
-                organizationName: bootstrapResource.settings.organizationName,
-                profileUri: bootstrapResource.accountSetting.links.self,
-            };
-        });
+        const { bootstrapResource } = await this.getBootstrapResource();
```
Comment (kandl) [inline libs/api-client-bear/src/user.ts:167]: This change can imo cause some issues, I remember that sometimes, binding xhr module to some methods of the bear-client was necessary to make it work.

File: libs/api-client-bear/src/user.ts [hunk 2]
```diff
@@ -252,11 +215,9 @@ export class UserModule {
     /**
      * Returns the feature flags valid for the currently logged in user.
      */
-    public getFeatureFlags(): Promise<GdcUser.IFeatureFlags> {
-        return this.xhr
-            .get("/gdc/app/account/bootstrap")
-            .then((r: any) => r.getData())
-            .then((result: any) => result.bootstrapResource.current.featureFlags);
+    public async getFeatureFlags(): Promise<GdcUser.IFeatureFlags> {
+        const { bootstrapResource } = await this.getBootstrapResource();
```
Comment (kandl) [inline libs/api-client-bear/src/user.ts:219]: Similar change as mentioned above.

**Review by kandl — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2486

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2248/ : SUCCESS in 1m 13s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2537/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1360/ : SUCCESS in 19m 15s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9613/ : SUCCESS in 13m 32s

**Pull request merged**